source/jcli/cli.d

```diff
@@ -2,327 +2,1929 @@ module jcli.cli;
 
 import jcli;
 
-import std.algorithm;
 import std.stdio : writefln, writeln;
-import std.array;
+import std.meta;
 
-// Needs a complete rework.
-final class CommandLineInterface(Modules...)
+struct ExecuteCommandResult
 {
-    alias Tokenizer     = ArgTokenizer!(string[]);
-    alias bindArgument  = bindArgumentAcrossModules!Modules;
+    int exitCode;
+    /// is null if no exception occured
+    Exception exception;
+}
+
+ExecuteCommandResult executeCommand(T)(auto ref scope T command)
+{
+    try
+    {
+        static if (is(typeof(command.onExecute()) : int))
+        {
+            return typeof(return)(command.onExecute(), null);
+        }
+        else static if (is(typeof(T.onExecute()) == void))
+        {
+            command.onExecute();
+            return typeof(return)(0, null);
+        }
+        else
+        {
+            return typeof(return)(0, null);
```

Comment:
Should this branch become a compile-time error, instead of silently failing to call onExecute?

My main concern is someone managing to mess up their definition of onExecute and getting no feedback from JCLI one what went wrong.

Reply:
The user may want to hook their command into another command, such that the next command depends on the logic of the intermediate command. So I think it's ok, given we go bottom-up (`ParentCommand`) instead of top-down (manually listing children). I think this should only be an error when the children were listed explicitly, which I have not implemented.

Reply:
@BradleyChatha sorry, misread that.
Noe that I have added a separate callback for when it's executed as an intermediate command, we can indeed make that an error, or default to printing the help of children, if there are any.